lib/cm/client.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Created by kugajjar on 6/3/15.
+ */
+
+'use strict';
+
+var crypto          = require('crypto'),
+    request         = require('request'),
+    HttpStatus      = require('http-status-codes'),
+    url             = require('url'),
+    Q               = require('q');
+
+module.exports = function client(nemo) {
+
+    var config = nemo._config.get('cm');
+
+    var generateToken = function (timeStamp) {
+
+        var secret = config.password + timeStamp;
+
+        return crypto.createHash('sha1')
+                     .update(secret)
+                     .digest('hex');
+    };
+
+    var authentication_params = function (){
```

Comment:
I'd rename this to `authenticationParams` just so it's consistent with other names.